metricbeat/module/elasticsearch/node_stats/data_xpack.go

```diff
@@ -0,0 +1,225 @@
+					}),
+					"old": c.Dict("young", s.Schema{
+						"collection_count":          c.Int("collection_count"),
+						"collection_time_in_millis": c.Int("collection_time_in_millis"),
+					}),
+				}),
+			}),
+		}),
+		"thread_pool": c.Dict("thread_pool", s.Schema{
+			"bulk": c.Dict("bulk", s.Schema{
+				"threads":  c.Int("threads"),
+				"queue":    c.Int("queue"),
+				"rejected": c.Int("rejected"),
+			}),
+			"generic": c.Dict("generic", s.Schema{
+				"threads":  c.Int("threads"),
+				"queue":    c.Int("queue"),
+				"rejected": c.Int("rejected"),
+			}),
+			"get": c.Dict("get", s.Schema{
+				"threads":  c.Int("threads"),
+				"queue":    c.Int("queue"),
+				"rejected": c.Int("rejected"),
+			}),
+			"index": c.Dict("index", s.Schema{
+				"threads":  c.Int("threads"),
+				"queue":    c.Int("queue"),
+				"rejected": c.Int("rejected"),
+			}),
+			"management": c.Dict("management", s.Schema{
+				"threads":  c.Int("threads"),
+				"queue":    c.Int("queue"),
+				"rejected": c.Int("rejected"),
+			}),
+			"search": c.Dict("search", s.Schema{
+				"threads":  c.Int("threads"),
+				"queue":    c.Int("queue"),
+				"rejected": c.Int("rejected"),
+			}),
+			"watcher": c.Dict("watcher", s.Schema{
+				"threads":  c.Int("threads"),
+				"queue":    c.Int("queue"),
+				"rejected": c.Int("rejected"),
+			}),
+		}),
+		"fs": c.Dict("fs", s.Schema{
+			"summary": c.Dict("total", s.Schema{
+				"total_in_bytes":     c.Int("total_in_bytes"),
+				"free_in_bytes":      c.Int("free_in_bytes"),
+				"available_in_bytes": c.Int("available_in_bytes"),
+			}),
+		}),
+	}
+)
+
+func eventsMappingXPack(r mb.ReporterV2, m *MetricSet, content []byte) {
+	nodesStruct := struct {
+		ClusterName string                            `json:"cluster_name"`
+		Nodes       map[string]map[string]interface{} `json:"nodes"`
+	}{}
+
+	json.Unmarshal(content, &nodesStruct)
+
+	// Normally the nodeStruct should only contain one node. But if _local is removed
+	// from the path and Metricbeat is not installed on the same machine as the node
+	// it will provid the data for multiple nodes. This will mean the detection of the
+	// master node will not be accurate anymore as often in these cases a proxy is in front
+	// of ES and it's not know if the request will be routed to the same node as before.
+	for nodeId, node := range nodesStruct.Nodes {
+		clusterId, err := elasticsearch.GetClusterId(m.http, m.HostData().SanitizedURI, nodeId)
+		if err != nil {
+			logp.Err("could not fetch cluster id: %s", err)
+			continue
+		}
+
+		isMaster, _ := elasticsearch.IsMaster(m.http, m.HostData().SanitizedURI)
+
+		event := mb.Event{}
+		// Build source_node object
+		sourceNode, _ := sourceNodeXpack.Apply(node)
+		sourceNode["uuid"] = nodeId
+
+		nodeData, _ := schemaXpack.Apply(node)
+		nodeStats := common.MapStr{
+			// This value is currently hard coded as it's not available in node_stats endpoint
+			"mlockall":    false,
```

Comment:
We can ignore this value, which was a holdover from when it was in `_nodes/stats`. We do not use it and it should be in an information document [that does not exist] -- not a stats one.

In case you're wondering, you _could_ fetch it via `GET /_nodes/_local/process`.

Reply:
Removed.